cpp/src/arrow/array/data.cc

```diff
@@ -208,7 +215,6 @@ int64_t ArraySpan::GetNullCount() const {
 int GetNumBuffers(const DataType& type) {
   switch (type.id()) {
     case Type::NA:
-      return 0;
```

Comment:
Is this correct?

Reply:
Yes — if you look at `NullType::layout()` there's a single buffer that's always null, but this seems to conflict with the columnar format documentation. @pitrou do you know more about this?

Reply:
Hmm, I suppose it's like how here we have unions listed with three buffers even though the spec lists only one/two.

Reply:
I think that was to avoid breaking compatibility. We noticed too late that this was inconsistent with the format spec.
We bridge for the inconsistency in the IPC and C data layers.